sirepo/package_data/static/js/cloudmc.js

```diff
@@ -2191,54 +2197,82 @@ SIREPO.app.directive('multiLevelEditor', function(appState, panelState) {
 SIREPO.app.directive('minMax', function(validationService) {
     return {
         require: 'ngModel',
         restrict: 'A',
         scope: {
             model: '=',
-            modelName: '=',
-            field: '=',
-            fieldName: '=',
-            info: '=',
+            modelName: '<',
+            field: '<',
+            min: '<',
+            max: '<',
         },
         template: `
-            <div data-ng-repeat="v in field.val track by $index" style="display: inline-block;">
-                <label data-text-with-math="field.labels[$index]" style="margin-right: 1ex"></label>
-                <button type="button" class="btn sr-button-action btn-xs" title="{{ globalButton($index).title }}" data-ng-click="toGlobal($index)"><span class="{{ globalButton($index).class }}"></span></button>
-                <input class="form-control sr-number-list" data-string-to-number="Float" data-ng-model="field.val[$index]" style="text-align: right" required />
+            <div data-ng-repeat="v in model[field] track by $index" style="display: inline-block;">
+                <label style="margin-right: 1ex">{{ globalButton[$index].label }}</label>
+                <button type="button" class="btn sr-button-action btn-xs" title="{{ globalButton[$index].title }}" data-ng-click="toGlobal($index)"><span class="{{ globalButton[$index].class }}"></span></button>
+                <input class="form-control sr-number-list" data-string-to-number="Float" data-ng-model="model[field][$index]" style="text-align: right" required />
                 <div data-ng-if="$last" class="sr-input-warning"></div>
             </div>
         `,
         controller: function($scope) {
             $scope.toGlobal = (index) => {
-                $scope.field.val[index] = $scope.field.global[index];
+                $scope.model[$scope.field][index] = index ? $scope.max : $scope.min;
             };
 
-            $scope.globalButton = (index) => [
+            $scope.globalButton = [
                 {
-                    title: `Set to global minimum (${$scope.field.global[0]})`,
+                    label: "Lower",
+                    title: "Set to global minimum",
                     class: 'glyphicon glyphicon-step-backward',
                 },
                 {
-                    title: `Set to global maximum ${$scope.field.global[1]})`,
+                    label: "Upper",
+                    title: "Set to global maximum",
                     class: 'glyphicon glyphicon-step-forward',
                 },
-            ][index];
+            ];
         },
         link: function(scope, element, attr, ngModel) {
             function validate() {
-                const t = scope.field.val;
-                const hasVals = ! t.some(x => x == null);
+                const t = scope.model[scope.field];
+                if (! t) {
+                    return;
+                }
+                const err = t.some(x => x === null || x === undefined)
```

Comment:
Do we have a policy about nesting ternary expressions? Personally hard for me to read, but that could just be me.

Reply:
I don't think they should be indented like that:

```js
                const err = t.some(x => x === null || x === undefined)
                    ? 'Enter values'
                    : t[0] === 0 && t[1] === 0
                    ? ''
                    : t[0] >= t[1]
                    ? 'Lower limit must be less than upper limit'
                    : '';
```js

Part of the problem is `t`. If `t[0]` and `t[1]` were something like t.lower and t.upper it would be clearer.

Reply:
I'll change the indent. I agree that named fields would be better. In this case it would be easier to have individual model fields for the lower and upper limit, ex. energyFilter.lowerLimit and energyFilter.upperLimit.